flake.nix

```diff
@@ -0,0 +1,32 @@
+{
+  description = "DockYard academy";
+
+  inputs = {
+      nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+      flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }: 
+    flake-utils.lib.eachDefaultSystem (system:
+      let pkgs = nixpkgs.legacyPackages.${system}; in {
+        devShell = pkgs.mkShell {
+          buildInputs = with pkgs; [
+            erlang
+            elixir
+          ];
+          shellHook = ''
+              mkdir -p .nix_mix
+              mkdir -p .nix_hex
+              export MIX_HOME=$PWD/utils/.nix_mix
+              export HEX_HOME=$PWD/utils/.nix_hex
+              export PATH=$MIX_HOME/bin:$PATH
+              export PATH=$HEX_HOME/bin:$PATH
+              export PATH=$MIX_HOME/escripts:$PATH
+              export LIVEBOOK_DEFAULT_RUNTIME=mix:$PWD/utils
```

Comment:
Does this cause issues with installing additional dependencies?

Many Livebooks install extra projects in the setup section, and if we use a default runtime I don't think we can install those dependencies.

Reply:
I set the default runtime for my convenience but seems like its not a good fit for this, I'll remove it so theres no confusion.